lms/djangoapps/courseware/masquerade.py

```diff
@@ -40,6 +40,16 @@ def __init__(self, course_key, role='student', user_partition_id=None, group_id=
         self.group_id = group_id
         self.user_name = user_name
 
+    def __setstate__(self, state):
+        """
+        Ensure that all attributes are initialised when unpickling CourseMasquerade objects.
+
+        Users might still have CourseMasquerade objects from older versions of the code in their
+        session.  These old objects might not have all attributes set, possibly resulting in
+        AttributeErrors.
+        """
+        self.__init__(**state)
```

Comment:
thanks for the explanation, do you mind adding a comment in case there is a non-backwards compatible schema change to the object's schema?